### PR TITLE
configures production as explained in guides for Heroku

### PR DESCRIPTION
**Pull request by jaimeiniesta**
Hi! Related to https://github.com/elixirschool/school_house/issues/22 I think I've found the cause, it looks like `config/prod.exs` needed some attention as per [this guide](https://hexdocs.pm/phoenix/heroku.html#making-our-project-ready-for-heroku).

### CODE REVIEW

**Comment by doomspork** [comment]
Howdy @jaimeiniesta! :tada: I think we want _some_ of this PR but not all of it. Some of the production configuration occurs at runtime in the `releases.exs` file here: https://github.com/elixirschool/school_house/blob/master/config/releases.exs#L17-L25

I had to remove the `force_ssl: [rewrite_on: [:x_forwarded_proto]],` option because of an issue between Cloudflare, Heroku, and Phoenix. With the current configuration between them all, this option results in an infinite loop redirecting to SSL. (Termination happens before Phoenix).

I'll revisit the Cloudflare rules and see if that's the culprit. Until I can do that we'll need to drop that line.

**Comment by jaimeiniesta** [comment]
Sounds good to me!

File: config/prod.exs [hunk 1]
```diff
@@ -10,7 +10,9 @@ use Mix.Config
 # which you should run after static files are built and
 # before starting your production server.
 config :school_house, SchoolHouseWeb.Endpoint,
-  url: [host: "example.com", port: 80],
+  http: [port: {:system, "PORT"}],
+  url: [scheme: "https", host: "beta.elixir.com", port: 443],
+  force_ssl: [rewrite_on: [:x_forwarded_proto]],
```
Comment (doomspork) [inline config/prod.exs:15]:
```suggestion
  url: [scheme: "https", host: "beta.elixirschool.com", port: 443]
```

**Review by doomspork — CHANGES_REQUESTED** [review]
@jaimeiniesta with this suggested change I think we should be safe to merge. The domain needed to be updated and we can't use the `force_ssl` option until I can clean up Cloudflare which appears to impact some other things so may take a little more effort.

**Comment by jaimeiniesta** [comment]
LGTM

File: config/prod.exs [hunk 2]
```diff
@@ -10,7 +10,8 @@ use Mix.Config
 # which you should run after static files are built and
 # before starting your production server.
 config :school_house, SchoolHouseWeb.Endpoint,
-  url: [host: "example.com", port: 80],
+  http: [port: {:system, "PORT"}],
+  url: [scheme: "https", host: "beta.elixirschool.com", port: 443]
```
Comment (doomspork) [inline config/prod.exs:14]:
```suggestion
  url: [scheme: "https", host: "beta.elixirschool.com", port: 443],
```

**Review by doomspork — CHANGES_REQUESTED** [review]

File: config/prod.exs (hunk 2 above)
Comment (doomspork) [reply config/prod.exs]: Round about way to get formatters changes onto a branch you don't have access to :tada:

**Review by doomspork — APPROVED** [review]

**Comment by doomspork** [comment]
I need to look into Heroku pipelines for testing out PRs. I'll try to carve out time this week or next to do that.

**Pull request merged**